src/celeritas/field/FieldPropagator.hh

```diff
@@ -260,7 +260,6 @@ CELER_FUNCTION auto FieldPropagator<DriverT>::operator()(real_type step)
             // to the extra delta_intersection boost when searching. The
             // substep itself can be more than the requested step.
             result.distance += celeritas::min(update_length, substep.step);
-            CELER_ASSERT(result.distance <= step);
```

Comment:
I also had a commit planned to remove this assertion :) but then reconsidered, because I think it can be more than just machine epsilon roundoff: the update length can be a bit more than the remaining step distance, and the accumulated substep can also be more than $\sim\epsilon$... (I don't think I've quantified how far off).

I think you're right that since the condition for changing the `step_limit` is effectively:
```c++
(!p.looping && !p.boundary && p.distance < step_limit.step)
```
we don't need to worry about returning slightly more than the input step.